gobblin-compaction/src/main/java/org/apache/gobblin/compaction/suite/CompactionSuiteBase.java

```diff
@@ -115,7 +115,7 @@ public FileSystemDataset load (final State state) {
    * @return  A list of {@link CompactionCompleteAction}s which needs to be executed after
    *          map-reduce is done.
    */
-  public List<CompactionCompleteAction<FileSystemDataset>> getCompactionCompleteActions() {
+  public List<CompactionCompleteAction<FileSystemDataset>> getCompactionCompleteActions() throws IOException {
     ArrayList<CompactionCompleteAction<FileSystemDataset>> array = new ArrayList<>();
```

Comment:
This is not your code but I found it really weird to call this `array`. Can you fix it as part of your PR ? just renaming to arrayList for example is good enough.